src/linker/Linker/LinkContext.cs

```diff
@@ -466,6 +469,26 @@ public AssemblyDefinition GetLoadedAssembly (string name)
 			return null;
 		}
 
+		public IEnumerable<AssemblyDefinition> ReferencedAssemblies() {
+			var assemblies = GetAssemblies ();
+
+			foreach (var assembly in assemblies)
+				yield return assembly;
+
+			var loaded = new HashSet<AssemblyDefinition> (assemblies);
+			var toProcess = new Queue<AssemblyDefinition> (assemblies);
+
+			while (toProcess.Count > 0) {
+				var assembly = toProcess.Dequeue ();
+				foreach (var reference in ResolveReferences (assembly)) {
+					if (!loaded.Add (reference))
+						continue;
+					yield return reference;
+					toProcess.Enqueue (reference);
+				}
+			}
```

Comment:
I'm not sure about this, which code does need all expanded references

Reply:
I'm still cleaning this up, but currently:
- SweepStep does this to handle some edge cases (hoping to get rid of this in the common case)
- it's used to match the current behavior of type name resolution which can find unqualified type names in any assembly
- XML that uses '*' to select all assemblies

Reply:
> XML that uses '*' to select all assemblies

We will need a different solution for this unfortunately. Especially if we rely on this mechanism to remove nullable annotations. We would need something like: register a callback which is called on all assemblies - already loaded and those in the future. And in it apply the necessary stuff from the XML.

We can definitely solve this in a followup change - but we should keep track of these things. Can you please create a tracking issue (or reuse existing one) and add the check mark list of things which need to happen before we are confident that we got everything we needed done.

I added a test for this case here: https://github.com/mono/linker/pull/1705

Reply:
Thanks, I forgot that we will always hit this case because of the corelib descriptors. I'll track it in https://github.com/mono/linker/issues/1708.

Reply:
Oh, and the expanded references are also needed for they `copy/save` action: https://github.com/mono/linker/pull/1666/commits/b3fef48e10896c2506abfe9d258ad1d9afe910f8#diff-f3ab7d627296ba105613b9cc039ca6f4ddc7a9aa66c6060ca82f6456ae0ede4fR416.